kai/reactive_codeplanner/agent/analyzer_fix/agent.py

```diff
@@ -131,7 +131,7 @@ def execute(self, ask: AgentRequest) -> AnalyzerFixResponse:
         resp = self.parse_llm_response(aimessage)
         return AnalyzerFixResponse(
             encountered_errors=[],
-            file_to_modify=ask.file_path,
+            file_to_modify=ask.file_path.absolute(),
```

Comment:
I think this should be `.resolve()` not `.absolute()`. `.absolute()` doesn't transform `..` parts into references to the parent directory or resolve symbolic links:

```python
from pathlib import Path
import os

print(os.getcwd())
# /home

p = Path("../thing")

print(p.absolute())
# /home/../thing
print(p.resolve())
# /thing
```

Reply:
do we really want to resolve symlinks though?

Reply:
updated to use `os.path.abspath` to expand the relative path things like `..` but not resolve symlinks.